runtime-light/stdlib/server/args-functions.cpp

```diff
@@ -0,0 +1,91 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2025 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#include <cctype>
+#include <cstdint>
+#include <string_view>
+
+#include "runtime-common/core/runtime-core.h"
+#include "runtime-light/state/instance-state.h"
+#include "runtime-light/stdlib/server/args-functions.h"
+
+Optional<array<mixed>> f$getopt(const string &short_options, const array<string> &long_options,
+                                [[maybe_unused]] Optional<std::optional<std::reference_wrapper<string>>> rest_index) noexcept {
+  if (const auto &instance_st{InstanceState::get()}; instance_st.image_kind() != ImageKind::CLI) [[unlikely]] {
+    return false;
+  }
+
+  enum class OptionKind : uint8_t { flag, required, optional };
+
+  const auto &cli_opts{ComponentState::get().cli_opts};
+  array<mixed> options;
+
+  std::string_view short_options_view{short_options.c_str(), short_options.size()};
+  // parse short options
+  for (size_t pos = 0; pos < short_options_view.size(); ++pos) {
+    if (!std::isalnum(short_options_view[pos])) {
+      continue;
+    }
+    const string option{1, short_options_view[pos]};
+
+    OptionKind kind{OptionKind::flag};
+    // check that char followed by a colon
+    if (pos + 1 < short_options_view.size() && short_options_view[pos + 1] == ':') {
+      kind = OptionKind::required;
+      pos++;
+
+      // check that char followed by a two colon
+      if (pos + 1 < short_options_view.size() && short_options_view[pos + 1] == ':') {
+        kind = OptionKind::optional;
+        pos++;
+      }
+    }
+
+    if (!cli_opts.has_key(option)) {
+      // option has not been set
+      continue;
+    }
+
+    const mixed &value{cli_opts.get_value(option)};
+    if (kind == OptionKind::optional) {
+      options.set_value(option, value.empty() ? false : value);
+    } else if (kind == OptionKind::required && !value.empty()) {
```

Comment:
shouldn't it be an error or warning that we don't have required option?

Reply:
In php where is no warnings on such cases